jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/milestones/CaseMilestoneItemView.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.console.ng.cm.client.milestones;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import com.google.gwt.user.client.TakesValue;
+import org.jboss.errai.common.client.api.IsElement;
+import org.jboss.errai.common.client.dom.Div;
+import org.jboss.errai.common.client.dom.HTMLElement;
+import org.jboss.errai.common.client.dom.Span;
+
+import org.jboss.errai.databinding.client.api.DataBinder;
+import org.jboss.errai.ui.shared.api.annotations.AutoBound;
+import org.jboss.errai.ui.shared.api.annotations.Bound;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.jbpm.console.ng.cm.client.util.AbstractView;
+import org.jbpm.console.ng.cm.client.util.CaseMilestoneStatusConverter;
+import org.jbpm.console.ng.cm.client.util.CaseMilestoneStatusEnum;
+import org.jbpm.console.ng.cm.client.util.DateConverter;
+import org.jbpm.console.ng.cm.model.CaseMilestoneSummary;
+
+import static org.jboss.errai.common.client.dom.DOMUtil.*;
+
+@Dependent
+@Templated
+public class CaseMilestoneItemView extends AbstractView<CaseMilestoneListPresenter> implements TakesValue<CaseMilestoneSummary>, IsElement {
+
+    @Inject
+    @DataField("milestone-name")
+    @Bound
+    Span name;
+
+    @Inject
+    @DataField("milestone-identifier")
+    @Bound
+    Span identifier;
+
+    @Inject
+    @DataField("milestone-status")
+    @Bound(converter = CaseMilestoneStatusConverter.class)
+    Span status;
+
+    @Inject
+    @DataField("list-group-item")
+    Div listGroupItem;
+    
+    @Inject
+    @AutoBound
+    private DataBinder<CaseMilestoneSummary> caseMilestoneSummary;
+
+    @Override
+    public HTMLElement getElement() {
+        return listGroupItem;
+    }
+
+    @Override
+    public CaseMilestoneSummary getValue() {
+        return caseMilestoneSummary.getModel();
+    }
+
+    @Override
+    public void setValue(final CaseMilestoneSummary model) {
+        this.caseMilestoneSummary.setModel(model);
+        final CaseMilestoneStatusEnum milestoneStatus = CaseMilestoneStatusEnum.fromStatus(model.getStatus());
+
+        if(milestoneStatus == CaseMilestoneStatusEnum.AVAILABLE){
+            String availableStr = new CaseMilestoneStatusConverter().toWidgetValue(model.getStatus());
```

Comment:
Also you can use switch on enum value here to make the code a bit more readable.

switch(milestoneStatus) {
   case AVAILABLE : ...; break;
   case COMPLETED: ...; break;
   case TERMINATED:
   default: ...
}